velox/type/DecimalUtil.h

```diff
@@ -356,6 +370,39 @@ class DecimalUtil {
         *writePosition++ = '-';
         unscaledValue = -unscaledValue;
       }
+      if (isScientific) {
```

Comment:
Could you use unscaledValue <1e-6 to check if the number should be scientific format, then we would not need countDigits for other common normal numbers